app/src/main/java/fr/free/nrw/commons/contributions/ContributionViewHolder.java

```diff
@@ -99,6 +106,30 @@ public void init(int position, Contribution contribution) {
         }
     }
 
+    /**
+     * In contributions first we show the title for the image stored in cache,
+     * then we fetch captions associated with the image and replace title on the thumbnail with caption
+     *
+     * @param contribution
+     */
+    private void fetchAndDisplayCaption(Contribution contribution) {
+        if ((contribution.getState() != Contribution.STATE_COMPLETED)) {
+            titleView.setText(contribution.getDisplayTitle());
+        } else {
+            Timber.d("Fetching caption for %s", contribution.getFilename());
+            String wikibaseMediaId = "M"+contribution.getPageId(); // Create Wikibase media id from the page id. Example media id: M80618155 for https://commons.wikimedia.org/wiki/File:Tantanmen.jpeg with has the pageid 80618155
+            compositeDisposable.add(mediaClient.getCaptionByWikibaseIdentifier(wikibaseMediaId)
```

Comment:
Are we not saving this to db?

Reply:
No clue, the are certainly fields in media and contribution that use `depiction` & `caption` liberally but I haven't drilled down enough to verify this